src/Compilers/CSharp/Test/Emit/CodeGen/IndexAndRangeTests.cs

```diff
@@ -278,8 +582,8 @@ public static void Main()
         var result = M(s);
         Console.WriteLine(result);
     }
-    public static string M(MyString s) => s[..^2];
-}" + TestSources.StringWithIndexers, expectedOutput: "abcd");
+    public static string M(string s) => s[..^2];
+}", expectedOutput: "abcd");
```

Comment:
Couple more test suggestions:

- Ensure we bind to `int this[Index indexer, int blah = default]` when present. 
- Ensure we bind to `int this[T value]` using the indexer syntax when `T` is instantiated as `int`

Reply:
My impression was that we *weren't* supposed to use the constructed symbol, just the underlying. I have a test that verifies that. Should I reverse it?